scripts/sendpayment.py

```diff
@@ -155,7 +151,18 @@ def main():
             .format(exp_tx_fees_ratio))
 
     if options.makercount == 0 and not options.p2ep:
-        direct_send(wallet, amount, mixdepth, destaddr, options.answeryes)
+        # we run a direct send as a single synchronous operation,
+        # then quit, but must wait for wallet sync before starting.
+        def wait_to_send():
+            if not ws.synced:
+                return
+            wl.stop()
```

Comment:
just to make sure, does this actually work? `wl` is referenced here before being defined in the outer scope.

Reply:
I guess this is a Python question? I've run this on regtest and it worked, yes. I guess it works because there is no context in which that code is reached without `wl` being defined?

Reply:
Yes, this was purely a python question. I wasn't entirely sure if would behave like that.